src/grad.jl

```diff
@@ -3,7 +3,7 @@
 
 Approximate the gradient of `f` at `x` using `fdm`. Assumes that `f(x)` is scalar.
 """
-function grad(fdm, f, x::AbstractArray{T}) where T<:Real
+function grad(fdm, f, x::Array{T}) where T<:Real
```

Comment:
Maybe turn `Array` into `Vector` to be consistent with the below?
```suggestion
function grad(fdm, f, x::Vector{T}) where T<:Real
```